### ISSUE TITLE
Desktop file validation error

### ISSUE THREAD

**Issue by xvitaly** [telegramdesktop/tdesktop#23879]
### Steps to reproduce

Run validation against telegramdesktop.desktop.

### Expected behaviour

Successful validation.

### Actual behaviour

```
$ desktop-file-validate telegramdesktop.desktop
telegramdesktop.desktop: error: file contains key "SingleMainWindow" in group "Desktop Entry", but keys extending the format should start with "X-"
```

Regression introduced by 4965c19314db73c248c2ae05b9dc06c8164bea2f.

`SingleMainWindow` should be removed until all DEs and validation tools will start supporting it.

### Operating system

Fedora 35

### Version of Telegram Desktop

master

### Installation source

Other (unofficial) source

**Comment by mymedia2** [comment]
> SingleMainWindow should be removed until all DEs and validation tools will start supporting it.

I do not agree with you. It worth having both [standardized][1] key along with prefixed one for faster transiting.
The tool you mentioned should be aware of the new key. And that is where the issue should be fixed.

 [1]: https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#recognized-keys

**Comment by xvitaly** [comment]
>  The tool you mentioned should be aware of the new key. And that is where the issue should be fixed.

1. Desktop SPEC version must be bumped to 1.5.
2. Most of GNU/Linux distributions and Flatpak require desktop file validation. Now maintainers will need to manually patch this row.

**Comment by mymedia2** [comment]
Well, the issue may be rephrased a bit. Something like "Incorrect spec version in .desktop file" would be more relevant. I do not think we have to abandon new features only because maintainers are under pressure from the old validator.

**Comment by xvitaly** [comment]
> I do not think we have to abandon new features only because maintainers are under pressure from the old validator.

We can add this line back when https://gitlab.freedesktop.org/xdg/desktop-file-utils/-/issues/59 will be resolved.

**Comment by ilya-fedin** [comment]
I think this should be patched by affected people until their tools are fixed. There's no need to remove that line upstream. Some people may opt to disabling the validation.

**Comment by xvitaly** [comment]
> I think this should be patched by affected people until their tools are fixed.

The official XDG validator doesn't support 1.5 schema yet: https://gitlab.freedesktop.org/xdg/desktop-file-utils/-/issues/59

> I think this should be patched by affected people until their tools are fixed.

In Flatpak too?

> Some people may opt to disabling the validation.

We can't. Validations of desktop and metainfo are mandatory and fully automatic.

**Comment by ilya-fedin** [comment]
> The official XDG validator doesn't support 1.5 schema yet

I don't see a problem here. We can be faster.

> In Flatpak too?

Why not?

> We can't. Validations of desktop and metainfo are mandatory.

You can always patch until they add the support.

**Comment by mikhailnov** [comment]
Which distros require this validation? I have not heard of any. You can simply remove this validation from your spec file, it is not done automatically, is it? Even if it is, you can patch the desktop file if validation by machine validators is more important than user experience _for you_.

**Comment by xvitaly** [comment]
> Which distros require this validation?

Fedora:

> It is not simply enough to just include the .desktop file in the package, one MUST run desktop-file-install (in %install) OR desktop-file-validate (in %check or %install) and have BuildRequires: desktop-file-utils, to help ensure .desktop file safety and spec-compliance.

**Comment by ilya-fedin** [comment]
Fedora is not the center of the universe

**Comment by xvitaly** [comment]
> Fedora is not the center of the universe

Flatpak.

**Comment by mikhailnov** [comment]
This does not make proposed deletion start making sense. Here you can either patch the desktop file or deal with packaging guidelines (btw you package telwgram npt for official repos)

**Comment by xvitaly** [comment]
OK. I will send a PR with SPEC version fix.

**Comment by mikhailnov** [comment]
> OK. I will send a PR with SPEC version fix.

Thanks! It will be not useless.

**Comment by gasinvein** [comment]
I think this was a sensible change that doesn't need to be reverted. But note that the `.desktop` file currently it truly invalid: it declares standard version `1.0`, which doesn't allow the `SingleMainWindow` key. What should be done here is changing `Version` to `1.5`; otherwise, the validation will fail even when the 1.5 spec support is added to desktop-file-utils.

**Comment by xvitaly** [comment]
Should be fixed in #23881.
